src/component/plotly-dash-preview/render.js

```diff
@@ -0,0 +1,65 @@
+const remote = require('../../util/remote')
+
+/**
+ * @param {object} info : info object
+ *  - url
+ *  - pdfOptions
+ * @param {object} opts : component options
+ * @param {function} sendToMain
+ *  - errorCode
+ *  - result
+ *    - imgData
+ */
+function render (info, opts, sendToMain) {
+  const result = {}
+
+  let win = remote.createBrowserWindow()
+  win.loadURL(info.url)
+
+  const contents = win.webContents
+
+  const done = errorCode => {
+    win.close()
+
+    if (errorCode) {
+      result.msg = 'dash preview generation failed'
+    }
+    sendToMain(errorCode, result)
+  }
+
+  /*
+   * We check for a 'waitfor' div in the dash-app
+   * which indicates that the app has finished rendering.
+   */
+  const finishedLoading = () => {
+    return win.webContents.executeJavaScript(`
+      new Promise((resolve, reject) => {
+        var a = document.getElementById("waitfor");
```

Comment:
Can we make this a variable? i.e. part of the API where the use can pass in an arbitrary CSS selector. That would modify this to be `document.querySelector`. 

For example, if the app has a graph in it, I might end up using a query selector like `#mygraph .plot-container` in order to wait until the graph was rendered (as plotly.js draws `.plot-container` dynamically). But, if the report doesn't have a graph in it, then I'd inject my own div with id `waitfor`

Reply:
Just to confirm, the query selector comes as the part of the incoming POST request (with `.plot-container` as default if nothing is provided)? 

It should be safe to assume the dash-app has atleast one chart even if the incoming request does not specify it.